src/keygen/participant.rs

```diff
@@ -665,9 +671,9 @@ mod tests {
             for _ in 0..quorum_size {
                 participant_ids.push(ParticipantIdentifier::random(rng));
             }
-            let participants = participant_ids
+            participant_ids
                 .iter()
-                .map(|&participant_id| -> KeygenParticipant {
+                .map(|&participant_id| -> Result<KeygenParticipant> {
```

Comment:
I see this pattern repeats. We should factor it out. We can have a general function on `ParticipantIdentifier`:
```rust
pub fn quorum<R: RngCore + CryptoRng>(
        size: usize,
        rng: &mut R,
    ) -> Vec<(ParticipantIdentifier, Vec<ParticipantIdentifier>)> {
        let participants: Vec<_> = (0..size)
            .map(|_| ParticipantIdentifier::random(rng))
            .collect();

        let mut quorum = Vec::new();
        for i in 0..size {
            let mut others = participants.clone();
            others.swap_remove(i);
            quorum.push((participants[i], others));
        }
        quorum
    }
```

Which creates the IDs. Then this function can just be:
```rust
pub fn new_quorum<R: RngCore + CryptoRng>(
            sid: Identifier,
            quorum_size: usize,
            rng: &mut R,
        ) -> Result<Vec<Self>> {
            let participants = ParticipantIdentifier::quorum(quorum_size, rng);
            participants
                .into_iter()
                .map(|(p, ps)| Self::new(sid, p, ps, ()))
                .collect()
        }
```

Reply:
Actually, this function already exists for `ParticipantConfig`, which is the set of participant IDs. I prefer it there because the single ID doesn't really need to understand the concept of quorums. But I'll refactor this and the equivalent in the auxinfo tests to call that function.